repo_update.sh

```
@@ -11,63 +11,141 @@ fi
 
 ANDROOT=$PWD
 
-pushd () {
+pushd() {
     command pushd "$@" > /dev/null
 }
 
-popd () {
-    command popd "$@" > /dev/null
+popd() {
+    command popd > /dev/null
+}
+
+enter_aosp_dir() {
+    LINK="$HTTP://android.googlesource.com/platform/${2:-$1}"
```

`set -e` is already used in the script,  to make the script more strict also let it fail in case a variable is not available has not been set (`set -eu`)

---

Done, had to fix our external arguments.

---

Just force-pushed this one because I forgot the `set +u` at the bottom. That's a bit sad tho, isn't there a better way to get the current state and restore that, or start a new "stack" and pop it off at the end? I don't feel like messing with parameters from the top level (since we essentially enforce `+eu` after termination...) And what happens if the script crashes/exists (exactly due to these flags), do we need an exit handler to reset it?